clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/provider/agent/AppEngineLoadBalancerCachingAgent.groovy

```diff
@@ -0,0 +1,239 @@
+import com.netflix.spinnaker.clouddriver.cache.OnDemandAgent
+import com.netflix.spinnaker.clouddriver.cache.OnDemandAgent.OnDemandResult
+import com.netflix.spinnaker.clouddriver.cache.OnDemandMetricsSupport
+import com.netflix.spinnaker.clouddriver.kubernetes.provider.view.MutableCacheData
+
+import groovy.util.logging.Slf4j
+
+import java.util.concurrent.TimeUnit
+
+import static com.netflix.spinnaker.cats.agent.AgentDataType.Authority.AUTHORITATIVE
+import static com.netflix.spinnaker.clouddriver.appengine.cache.Keys.Namespace.LOAD_BALANCERS
+import static com.netflix.spinnaker.clouddriver.appengine.cache.Keys.Namespace.ON_DEMAND
+
+@Slf4j
+class AppEngineLoadBalancerCachingAgent extends AbstractAppEngineCachingAgent implements OnDemandAgent {
+  final String category = "loadBalancer"
+
+  final OnDemandMetricsSupport metricsSupport
+
+  static final Set<AgentDataType> types = Collections.unmodifiableSet([
+    AUTHORITATIVE.forType(LOAD_BALANCERS.ns),
+  ] as Set)
+
+  String agentType = "${accountName}/${AppEngineLoadBalancerCachingAgent.simpleName}"
+
+  @Override
+  String getSimpleName() {
+    AppEngineLoadBalancerCachingAgent.simpleName
+  }
+
+  @Override
+  String getOnDemandAgentType() {
+    "${agentType}-OnDemand"
+  }
+
+  @Override
+  Collection<AgentDataType> getProvidedDataTypes() {
+    types
+  }
+
+  AppEngineLoadBalancerCachingAgent(String accountName,
+                                    AppEngineNamedAccountCredentials credentials,
+                                    ObjectMapper objectMapper,
+                                    Registry registry) {
+    super(accountName, objectMapper, credentials)
+    metricsSupport = new OnDemandMetricsSupport(
+      registry,
+      this,
+      "$AppEngineCloudProvider.ID:$OnDemandAgent.OnDemandType.LoadBalancer")
+  }
+
+  @Override
+  boolean handles(OnDemandAgent.OnDemandType type, String cloudProvider) {
+    type == OnDemandAgent.OnDemandType.LoadBalancer && cloudProvider == AppEngineCloudProvider.ID
+  }
+
+  @Override
+  OnDemandResult handle(ProviderCache providerCache, Map<String, ? extends Object> data) {
+    if (!data.containsKey("loadBalancerName") || data.account != accountName) {
+      return null
+    }
+
+    def loadBalancerName = data.loadBalancerName.toString()
+
+    Service service = metricsSupport.readData {
+      loadService(loadBalancerName)
+    }
+
+    CacheResult result = metricsSupport.transformData {
+      buildCacheResult([service], [:], [], Long.MAX_VALUE)
+    }
+
+    def jsonResult = objectMapper.writeValueAsString(result.cacheResults)
+    def loadBalancerKey = Keys.getLoadBalancerKey(accountName, loadBalancerName)
+    if (result.cacheResults.values().flatten().isEmpty()) {
+      providerCache.evictDeletedItems(ON_DEMAND.ns, [loadBalancerKey])
+    } else {
+      metricsSupport.onDemandStore {
+        def cacheData = new DefaultCacheData(
+          loadBalancerKey,
+          TimeUnit.MINUTES.toSeconds(10) as Integer,
```

Comment:
Not a big deal, but you'll incur some extra conversion cost here. The `DefaultCacheData` constructor takes an `int`. If you cast it to an `Integer`, java will first autobox it for you into that `Integer`, then it will unbox it for you into the `int` the constructor requires. Should be able to just use `as int`.